ReText/window.py

```diff
@@ -1161,23 +1161,48 @@ def insertChars(self, chars):
 		else:
 			tc.insertText(chars)
 
+	# 	ut can be 'header', 'italic', 'bold', 'numbering', 'bullets', 'image', 'link', 'code', 'blockquote'
 	def insertTag(self, ut):
 		if not ut:
 			return
 		if isinstance(ut, int):
 			ut = self.usefulTags[ut - 1]
-		arg = ' style=""' if ut == 'span' else ''
+
 		tc = self.editBoxes[self.ind].textCursor()
-		if ut == 'img':
-			toinsert = ('<a href="' + tc.selectedText() +
-			'" target="_blank"><img src="' + tc.selectedText() + '"/></a>')
-		elif ut == 'a':
-			toinsert = ('<a href="' + tc.selectedText() +
-			'" target="_blank">' + tc.selectedText() + '</a>')
+
+		if ut == 'header':
+			toinsert = ('# ' + tc.selectedText())
+		elif ut == 'italic':
+			toinsert = ('*' + tc.selectedText() + '*')
+		elif ut == 'bold':
+			toinsert = ('**' + tc.selectedText() + '**')
+		elif ut == 'numbering':
+			toinsert = ('\n * ' + tc.selectedText())
+		elif ut == 'bullets':
+			toinsert = ('\n 1. ' + tc.selectedText())
+		elif ut == 'image':
+			selectedText = tc.selectedText()
+			if not selectedText:
+				selectedText = 'alt text'
+			toinsert = ('!['+ selectedText +'](url)')
+		elif ut == 'link':
+			selectedText = tc.selectedText()
+			if not selectedText:
+				selectedText = 'alt text'
+			toinsert = ('['+ selectedText +'](url)')
+		elif ut == 'code':
+			selectedText = tc.selectedText()
+			if not selectedText:
+				toinsert = '\n\n\t'
+			else: 
+				toinsert = ('`'+ selectedText +'`')
```

Comment:
Please note that `code` command has two behaviour depending on whether or not text is selected.
I'm not sure about that:
 - it's comfortable having two result with one command
 - it could be confusing for the newbie